### DISCUSSION TITLE
mailer doesn't declare dependency on c7n

### DISCUSSION THREAD

**Discussion by jawadmrahman350** [cloud-custodian/cloud-custodian#7423] [General]
### Describe the bug

Following the documentation found online, I was trying to use the c7n_mailer:
https://cloudcustodian.io/docs/tools/c7n-mailer.html
I followed everything as mentioned in the documentation and I an error when running this command: 
c7n-mailer --config mailer.yml --update-lambda

Traceback (most recent call last):

  File "/usr/local/bin/c7n-mailer", line 5, in <module>

    from c7n_mailer.cli import main

  File "/usr/local/lib/python3.9/site-packages/c7n_mailer/cli.py", line 11, in <module>

    from c7n_mailer import deploy, utils

  File "/usr/local/lib/python3.9/site-packages/c7n_mailer/deploy.py", line 8, in <module>

    from c7n.mu import (

ModuleNotFoundError: No module named 'c7n'

However, I have c7n installed via pip3. Nothing changes.  I saw Issue #3275 and did not follow sudo ever, just followd whatever the documentation says. 

### What did you expect to happen?

I expected a lambda to be created and mail to be sent. 

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Custodian:   0.9.13

Python:      3.9.6 (default, Jun 29 2021, 06:20:32) 

             [Clang 12.0.0 (clang-1200.0.32.29)]

Platform:    posix.uname_result(sysname='Darwin', nodename='PEARSONML-V0RTHTD', release='19.6.0', version='Darwin Kernel Version 19.6.0: Tue Aug 24 20:28:00 PDT 2021; root:xnu-6153.141.40~1/RELEASE_X86_64', machine='x86_64')

Using venv:  True

Docker: False

Installed: 


argcomplete==1.12.3

attrs==21.2.0

boto3==1.17.102

botocore==1.20.102

importlib-metadata==4.6.0

jmespath==0.10.0

jsonschema==3.2.0

pyrsistent==0.18.0

python-dateutil==2.8.1

pyyaml==5.4.1

s3transfer==0.4.2

setuptools==58.2.0

six==1.16.0

tabulate==0.8.9

typing-extensions==3.10.0.0

urllib3==1.26.6

zipp==3.4.1
```

### Relevant log/traceback output

```shell
Traceback (most recent call last):

  File "/usr/local/bin/c7n-mailer", line 5, in <module>

    from c7n_mailer.cli import main

  File "/usr/local/lib/python3.9/site-packages/c7n_mailer/cli.py", line 11, in <module>

    from c7n_mailer import deploy, utils

  File "/usr/local/lib/python3.9/site-packages/c7n_mailer/deploy.py", line 8, in <module>

    from c7n.mu import (

ModuleNotFoundError: No module named 'c7n'
```

**Comment by kapilt** [comment]
this seems like an install issue, if you pip install c7n it should resolve

it does appear the root of the issue is that c7n-mailer doesn't declare its dependency on c7n

**Comment by jawadmrahman350** [comment]
I did already install c7n.

On Wed, Oct 6, 2021 at 11:51 AM Kapil Thangavelu ***@***.***>
wrote:

> this seems like an install issue, if you pip install c7n it should resolve
>
> it does appear the root of the issue is that c7n-mailer doesn't declare
> its dependency on c7n
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cloud-custodian/cloud-custodian/issues/6926#issuecomment-936673138>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFMU47OUJCEFXW2GVV6XFMLUFR475ANCNFSM5FKC4CGA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Md Jawad Mashrur Rahman
Computer Engineering
Iowa State University
